src/libs/Navigation/linkTo.ts

```diff
@@ -99,9 +186,54 @@ export default function linkTo(navigation: NavigationContainerRef<RootStackParam
             // If this action is navigating to the ModalNavigator and the last route on the root navigator is not already opened ModalNavigator then push
         } else if (isModalNavigator(action.payload.name) && !isTargetNavigatorOnTop) {
```

Comment:
Opening new workspace from workspace list is not creating new workspace, ref: https://github.com/Expensify/App/issues/38420